apps/token/src/components/token-input/token-input.tsx

```diff
@@ -0,0 +1,192 @@
+import './token-input.scss';
+
+import {
+  Button,
+  Callout,
+  Input,
+  Intent,
+  FormGroup,
+} from '@vegaprotocol/ui-toolkit';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import type {
+  TransactionAction,
+  TransactionState,
+} from '../../hooks/transaction-reducer';
+import {
+  TransactionActionType,
+  TxState,
+} from '../../hooks/transaction-reducer';
+import { BigNumber } from '../../lib/bignumber';
+// TODO: TFE Import
+import { TransactionCallout } from '../transaction-callout';
+
+const inputName = 'amount';
+
+export const AmountInput = ({
+  amount,
+  setAmount,
+  maximum,
+  currency,
+}: {
+  amount: string;
+  setAmount: React.Dispatch<any>;
+  maximum: BigNumber;
+  currency: string;
+}) => {
+  const { t } = useTranslation();
+  return (
+    <div className="token-input__container">
+      <Input
+        data-testid="token-amount-input"
+        className="token-input__input"
+        name={inputName}
+        id={inputName}
+        onChange={(e) => setAmount(e.target.value)}
+        value={amount}
+        // TODO: TFE import
```

Comment:
?

Reply:
For the time being we wont render the currency inside the input until Input from ui-toolkit can  accept generic elements to prepend/append. See issue:
https://github.com/vegaprotocol/frontend-monorepo/issues/273